tests/components/analytics/test_analytics.py

```diff
@@ -1426,3 +1440,264 @@ async def async_modify_analytics(
             },
         },
     }
+
+
+async def test_send_snapshot_disabled(
+    hass: HomeAssistant,
+    aioclient_mock: AiohttpClientMocker,
+) -> None:
+    """Test no snapshots are sent."""
+    analytics = Analytics(hass)
+
+    await analytics.send_snapshot()
+
+    await analytics.save_preferences({ATTR_BASE: False, ATTR_SNAPSHOTS: True})
+    await analytics.send_snapshot()
+
+    await analytics.save_preferences({ATTR_BASE: True, ATTR_SNAPSHOTS: False})
+    await analytics.send_snapshot()
+
+    assert len(aioclient_mock.mock_calls) == 0
+
+
+async def test_send_snapshot_success(
+    hass: HomeAssistant,
+    caplog: pytest.LogCaptureFixture,
+    aioclient_mock: AiohttpClientMocker,
+) -> None:
+    """Test successful snapshot submission."""
+    snapshot_endpoint = ANALYTICS_SNAPSHOT_ENDPOINT_URL
+    aioclient_mock.post(
+        snapshot_endpoint,
+        status=200,
+        json={"submission_identifier": "test-identifier-123"},
+    )
+    analytics = Analytics(hass)
+    await analytics.save_preferences({ATTR_BASE: True, ATTR_SNAPSHOTS: True})
+
+    await analytics.send_snapshot()
+
+    assert len(aioclient_mock.mock_calls) == 1
+    assert analytics._data.submission_identifier == "test-identifier-123"
+    assert "Submitted snapshot analytics to Home Assistant servers" in caplog.text
+
+
+async def test_send_snapshot_with_existing_identifier(
+    hass: HomeAssistant,
+    caplog: pytest.LogCaptureFixture,
+    aioclient_mock: AiohttpClientMocker,
+) -> None:
+    """Test snapshot submission with existing identifier."""
+    snapshot_endpoint = ANALYTICS_SNAPSHOT_ENDPOINT_URL
+    aioclient_mock.post(
+        snapshot_endpoint,
+        status=200,
+        json={"submission_identifier": "test-identifier-123"},
+    )
+    analytics = Analytics(hass)
+    analytics._data.submission_identifier = "old-identifier"
```

Comment:
can we mock `Store.async_load()` instead, so we avoid using internals of the class being tested? It would also make the `AnalyticsData.from_dict` covered.